### DISCUSSION TITLE
QFieldCloud: uploading project partially

### DISCUSSION THREAD

**Discussion by olegzhelezn** [opengisch/QField#2482] [Ideas]
Dear QField Team, 

my project contains a very large .mbtiles and two large .gpkg files. They are used only as non-editable features, so, there is no need for me to upload them in the cloud. So, my question is: is it possible to only upload the editable layers in the cloud, and keep non-editable just on the device? If yes, how? If no, this possibility would be really appreciated in the future versions!

**Comment by suricactus** [comment]
Hi @olegzhelezn . Thanks for checking QFieldCloud!

Currently this is possible to achieve this in two different situations.

- Once uploaded, a file will not be shown for update unless it was modified either locally, or on the cloud. (However, sometimes GeoPackage files show as available for upload because of the internal implementation of .gpkg format. We are currently looking for a solution there.).
- You can also choose what to do with a file on the sync dialog. If you uncheck the "upload" checkbox, it will not be uploaded. For example the higlighted blue row will not be uploaded.

![image](https://user-images.githubusercontent.com/2820439/152517983-c7af5169-55ea-4f4d-b691-8d050ad35ba8.png)


If you ask for a specific functionality that "remove from project" action is respected when uploading to the cloud - it would be nice feature that we would consider to implement at some point in the future.

**Reply by olegzhelezn** [reply]
Dear @suricactus, thanks for a quick reply!
However, I mean something slightly different. The thing is, my .mbtiles is around 5GB (it’s VHR satellite imagery for a large area), so, I am not sure it would overall be a good idea to upload it to the cloud initially, when creating a new cloud project. Basically, I only need a couple of layers to be synced, and everything else is a basemap (tiles, vector elevation lines, rivers, etc.)
So, is there a possibility to have project A containing [basemap layers + editable layers] on the mobile device, and project A* with only [editable] layers in the cloud, so that they still can work together?

**Reply by JakubCha** [reply]
Hello! 

@suricactus That's key question also on my side. Ability to split project data to:
- "basemap" data with very heavy raster  (and sometimes vector) layers, e.g. multiple aerial orthophotos. This data should NOT be uploaded to cloud and should NOT be synchronized between QField and cloud.
- data (vector layers on postgis database) that needs to be synchronized between multiple users nearly simultaneously

What do you think?

**Comment by suricactus** [comment]
hey, thanks you both for the details of your requests:

@olegzhelezn If I get you right, you want to share your info between multiple projects. This is a planned feature, so you can share settings and data across projects, but we are not there yet. If it happens, it will be in the latter part of the year.

@JakubCha you feature request is more of a `qfieldcloudignore` list (similar to `.gitignore`), where certain files are not synced by default. While this is an interesting feature, currently it is only supported for files/directories that are in the root project directory and start with a `.`.

Long story short, for now you can use only the initially proposed solution, the rest are further development that is not planned in the new 3 months.

**Reply by JakubCha** [reply]
Thank you for your reply @suricactus 
Could you provide some example or link to docs about how to use `qfieldcloudignore`, where to define it, etc.?

**Reply by suricactus** [reply]
@JakubCha probably I was not very clear, `qfieldcloudignore` is a feature that does not exist yet. As a workaround you can create a file `.hidden_data` in your root project directory and it will be ignored by QFieldSync BETA20 onwards (yet to be released).

**Comment by ZsanettMed** [comment]
Hi @olegzhelezn and @JakubCha!

There is a feature request for this on our Ideas Platform. We encourage you to upvote it, as user feedback greatly influences our development priorities. You can find the feature request [here](https://ideas.qfield.org/qfieldcloud-feature-requests/p/selective-file-versioning-for-storage-optimization).

P.S.: If your needs are different from what's described in the existing feature request, please feel free to add more requests on the [QField Ideas Platform](https://ideas.qfield.org/).

Best, 
Zsanett

**Reply by L10ns** [reply]
Hi @olegzhelezn and @JakubCha!
I created a feature request in Ideas Platform here for our case: [https://ideas.qfield.org/qfieldcloud-feature-requests/p/qfieldcloud-and-large-data-on-device].
Thanks @ZsanettMed, I didn't know about this beautiful idea of Qfield Ideas Platform!
